improver/regrid/landsea.py

```diff
@@ -311,6 +302,32 @@ def correct_where_input_true(self, selector_val: int) -> None:
             ),
             in_vicinity.data > 0.5,
         )
+        return mismatch_points, indices, use_points, no_use_points
+
+    def correct_where_input_true(self, selector_val: int) -> None:
+        """
+        Replace points in the output_cube where output_land matches the
+        selector_val and the input_land does not match, but has matching
+        points in the vicinity, with the nearest matching point in the
+        vicinity in the original nearest_cube.
+        Updates self.output_cube.data.
+
+        Args:
+            selector_val:
+                Value of mask to replace if needed.
+                Intended to be 1 for filling land points near the coast
+                and 0 for filling sea points near the coast.
+        """
+        try:
+            mismatch_points, indices, use_points, no_use_points = self._get_matches(
+                selector_val
+            )
+        except NoMatchesError:
+            return
+        selector_data = self.nearest_cube.data.copy()
+        selector_data[no_use_points[0], no_use_points[1]] = selector_data[
+            use_points[0], use_points[1]
+        ][indices]
```

Comment:
I think this could just be:
```suggestion
        selector_data[no_use_points] = selector_data[use_points][indices]
```

Reply:
Good spotting, thanks